scripts/generate

```diff
@@ -191,7 +191,9 @@ def('substitutions',
                           K(el('span', 'arrow', htmlEncode('~>')))],
                          [contains(_, ap([I, htmlEncode], ['->'])),
                           K(el('span', 'arrow',
-                               el('span', 'arrow-hyphen', htmlEncode('-')) +
+                               //  Use NON-BREAKING HYPHEN so arrow does
+                               //  not break because of line wrapping
+                               el('span', 'arrow-hyphen', '&#x2011;') +
```

Comment:
My reasoning for using HTML entity here instead of Unicode character is to avoid confusion in the future: visually non-breaking hyphen and hyphen-minus look the same in most fonts (especially monospaced ones), so someone might not notice the difference and edit it out by accident.